freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020  Red Hat, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from freshmaker import db
+from freshmaker.handlers import ContainerBuildHandler
+from freshmaker.events import BotasChangeStateEvent
+from freshmaker.models import Event
+from freshmaker.types import EventState
+
+
+class HandleBotasAdvisory(ContainerBuildHandler):
+    """
+    Handles event that was created by transition of an advisory filed by
+    BOTAS to SHIPPED_LIVE state
+    """
+    name = "HandleBotasAdvisory"
+    order = 0
+
+    def can_handle(self, event):
+        return isinstance(event, BotasChangeStateEvent)
+
+    def handle(self, event):
+        if event.dry_run:
+            self.force_dry_run()
+        self.event = event
+
+        # Generate the Database representation of `event`, it can be
+        # triggered by user, we want to track what happened
+        db_event = Event.get(db.session, event.msg_id)
+
+        if not db_event:
+            self.log_warn("Can't find in DB event with id {}".format(
+                event.msg_id))
+            return
+
+        db.session.commit()
+        self.set_context(db_event)
+
+        # Get original nvrs of all builds in the advisory
+        original_nvrs = [build.original_nvr for build in db_event.builds]
+        self.log_info("Orignial nvrs of build in the advisory #{0} are: {1}".format(
+            db_event.search_key, " ".join(original_nvrs)))
+
+        msg = "We are skipping that event, because we are blocked in further" \
+              " implementation for now."
+
+        # Skip that event because we can't proceed with processing it.
+        # TODO
+        # Next step would be queries Pyxis and getting the digests of these nvrs
+        db_event.transition(EventState.SKIPPED, msg)
+        db.session.commit()
+        return original_nvrs
```

Comment:
This method should return an empty list or a list of event objects to process:
https://github.com/redhat-exd-rebuilds/freshmaker/blob/40e36b671fe123f2e1427ef89b00fd4896f613f2/freshmaker/handlers/__init__.py#L255-L265
https://github.com/redhat-exd-rebuilds/freshmaker/blob/40e36b671fe123f2e1427ef89b00fd4896f613f2/freshmaker/consumer.py#L159-L174

FYI, this is a bad practice since the messages are added to a queue that lives in memory so they will survive a restart., but that's current architecture and there's nothing we can do about it right now.

Reply:
Ok, didn't see it. I thought the return value isn't used so I can use it for testing, I will remove it. Btw do you know some way how to test that 'original_nvrs' variable except logging?

Reply:
I can't really think of a good way. It's fine to keep that part untested for now since it's just a simple list comprehension. It'll just be tested when this handler is expanded.